lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -263,12 +278,19 @@ def configure(conf)
       # Functions.
       @running_cloudfunctions = false
 
-      # Set labels, etc. based on the config
+      # Set up the MonitoredResource, labels, etc. based on the config
       case @platform
       when Platform::GCE
-        @service_name = COMPUTE_SERVICE
+        @resource.type = COMPUTE_RESOURCE_TYPE
+        # TODO: introduce a new MonitoredResource-centric configuration and
+        # deprecate subservice-name; for now, translate known uses.
         if @subservice_name
-          @service_name = @subservice_name
+          # TODO: what should we do if we encounter an unknown value?
+          if @subservice_name == DATAFLOW_SERVICE
+            @resource.type = DATAFLOW_RESOURCE_TYPE
+          elsif @subservice_name == ML_SERVICE
+            @resource.type = ML_RESOURCE_TYPE
```

Comment:
One of the main motivations on our side is to make the v2 migration transparent to the user (beyond an agent upgrade). This means that their existing configs should continue working, and, in particular, service name overrides and label mappings should work as expected. Switching from one resource type per service to multiple resource types per service is an orthogonal change that we can make at a later point, when we're already writing to v2.